src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -2223,6 +2229,8 @@ private static void RegisterSerializationProviders(List<TypeInfo> providerTypes)
                 return;
             }
 
+            externalSerializers.Clear();
+            typeToExternalSerializerDictionary.Clear();
```

Comment:
if this was just for testing, maybe instead we should clear these when calling `InitializeForTesting` instead?